numpy/core/_methods.py

```diff
@@ -288,3 +289,8 @@ def _dump(self, file, protocol=2):
 
 def _dumps(self, protocol=2):
     return pickle.dumps(self, protocol=protocol)
+
+def _bit_count(a, out=None, *, where=True, casting='same_kind',
```

Comment:
``` python
def bitcount(
    self,
    out: None | NDArray[Any] = ...,
    *,
    where: _ArrayLikeBool_co = ...,
    casting: _CastingKind = ...,
    order: _OrderKACF = ...,
    dtype: DTypeLike = ...,
    subok: bool = ...,
) -> NDArray[Any]: ...  # Or `Any` in the case of `np.generic`
```
Could you also add some basic annotations for `ndarray.bitcount` and `generic.bitcount`?

Adding dtype support is a bit more involved here, so I'd suggest to just copy/paste the annotations above and 
reserve further improvements for a (potential) future PR.

Reply:
Sure @BvB93 , will add them.

Reply:
Should the output be `-> integer | NDArray[Any]: ...` or does NDArray cover the 0-d?

Reply:
Eh, 0D arrays are a very tricky subject with the current lack of shape-typing (https://github.com/numpy/numpy/issues/19064) so I'd stick to just `NDArray[Any]`.

The problem with returning a union is that the effective signature of these objects is based on the intersection of all of its its members, _i.e._ it would make all operations that aren't supported by both `integer` and `ndarray` illegal (like iteration). 
All in all this would result in a plethora of false-positive errors for the end user.

The most practical way of dealing with this situation is by lying, _i.e._ ignoring the existence of 0D arrays. 
It's unfortunate, but also the lesser of (three or so) evils.

Reply:
Hey @BvB93 , can we add it to numpy/typing/_ufunc.pyi under `_UFunc_Nin1_Nout1`?

Reply:
> Hey @BvB93 , can we add it to numpy/typing/_ufunc.pyi under `_UFunc_Nin1_Nout1`?

I'd just stick adding them as normal methods to `ndarray` and `generic`.